src/cmdstan/command.hpp

```diff
@@ -860,7 +565,7 @@ int command(int argc, const char *argv[]) {
         info(
             "The number of warmup samples (num_warmup) must be greater than "
             "zero if adaptation is enabled.");
-        return_code = stan::services::error_codes::CONFIG;
+        return_code = return_codes::NOT_OK;
```

Comment:
I agree that this is a good change for consistency, but I wonder if we should be moving toward trying to use more descriptive exit codes, rather than more generic ones

Reply:
the stan::services layer is implemented with a set of posix return codes, defined here:  https://github.com/stan-dev/stan/blob/develop/src/stan/services/error_codes.hpp.  these aren't used consistently across the services, and, in fact, some of them are not used at all.

the definitions in std::services::error_codes are only really appropriate for C++ layers.   if the user is working in the shell, they don't really care about the return code, only the error message.   if the command is executed programmatically, either from a script, or Python, R, Julia, etc, then the error code needs to be interpreted correctly by the interface.